packaging/irods_prc_tests/test_rule_engine_plugin_metadata_guard_atomic.py

```diff
@@ -0,0 +1,94 @@
+from irods.access import iRODSAccess
+from irods.session import iRODSSession
+from irods.meta import iRODSMeta, AVUOperation
+
+import irods.exception
+
+import json
+import os
+import shutil
+import tempfile
+import unittest
+
+class test_metadata_guard(unittest.TestCase):
+
+    IRODS_CONFIG_FILE_PATH = '/etc/irods/server_config.json'
+    TEST_DATA_OBJECT_PATH = '/tempZone/home/alice/test1'
+
+    @classmethod
+    def backup_irods_config(cls):
+        cls.temp_path = os.path.join(tempfile.mkdtemp(), 'irods_config_backup')
+        shutil.copy2(cls.IRODS_CONFIG_FILE_PATH, cls.temp_path)
+
+    @classmethod
+    def restore_irods_config(cls):
+        shutil.copy2(cls.temp_path,cls.IRODS_CONFIG_FILE_PATH)
+
+    @classmethod
+    def setUpClass(cls):
+        with iRODSSession(host='localhost', port=1247, user='rods', password='rods', zone='tempZone') as local_admin:
+            # create unprivileged user
+            local_admin.users.create_with_password('alice', 'test')
+
+            # unprivileged user creates test file
+            with iRODSSession(host='localhost', port=1247, user='alice', password='test', zone='tempZone') as unprivileged_user:
+                unprivileged_user.data_objects.create(cls.TEST_DATA_OBJECT_PATH)
+                unprivileged_user.acls.set( iRODSAccess('own', cls.TEST_DATA_OBJECT_PATH, 'rods', 'tempZone'))
```

Comment:
Does this need to happen between tests or just on the first run of the whole suite? Just making sure I'm following this right.

Reply:
I think this whole method happens only once when the class is set up.